package im_client;

message InitMessage {
    required string name = 1;
    required string secret = 2;
}

message InitResultMessage {
    enum InitErrorCode {
        IncorrectSecret = 0;
        AlreadyRegistered = 1;
        ExpectedInitMessage = 2;
    }
    enum InitResult {
        Success = 0;
        Error = 1;
    }
    required InitResult result = 1;
    optional InitErrorCode error = 2;
}

message SubscribeMessage {
    required string name = 1;
    optional bytes data = 2;
}